ee/server/service/mdm.go

```diff
@@ -902,12 +908,22 @@ func (svc *Service) mdmSSOHandleCallbackAuth(
 	}
 
 	err = svc.ds.InsertMDMIdPAccount(ctx, &fleet.MDMIdPAccount{
+		UUID:     ssoRequestData.HostUUID,
 		Username: username,
 		Fullname: auth.UserDisplayName(),
 		Email:    auth.UserID(),
 	})
```

Comment:
If `UUID` is not provided, we'll still create a random one (see `InsertMDMIdPAccount` changes below)

Reply:
I was mentioning this more because this is how macOS currently operates, where it creates a random UUID and then later reconciles it by passing that random UUID to the server during the mdm enrollment. So I was pointing out that the macOS use case will still work and that the UUID here is not required.

If we don't pass one for the Linux/Windows use case, we're kinda cooked, which is why [the enroll endpoint bails out](https://github.com/fleetdm/fleet/pull/34835/files) if it doesn't see a hardware UUID.

Reply:
Got it, thanks. Saw the bail in tha code in my initial review.